streams/src/test/java/org/apache/kafka/streams/integration/StandByTaskCreationIntegrationTest.java

```diff
@@ -0,0 +1,139 @@
+        streamsConfiguration.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+        return streamsConfiguration;
+    }
+
+    @Test
+    public void shouldNotCreateAnyStandByTasksForStateStoreWithLoggingDisabled() throws InterruptedException {
+        final StreamsBuilder builder = new StreamsBuilder();
+
+        final String stateStoreName = "myTransformState";
+        final StoreBuilder<KeyValueStore<Integer, Integer>> keyValueStoreBuilder =
+            Stores.keyValueStoreBuilder(Stores.persistentKeyValueStore(stateStoreName),
+                                        Serdes.Integer(),
+                                        Serdes.Integer()).withLoggingDisabled();
+        builder.addStateStore(keyValueStoreBuilder);
+
+        builder.stream(INPUT_TOPIC, Consumed.with(Serdes.Integer(), Serdes.Integer()))
+            .transform(() -> new Transformer<Integer, Integer, KeyValue<Integer, Integer>>() {
+                private KeyValueStore<Integer, Integer> state;
+
+                @SuppressWarnings("unchecked")
+                @Override
+                public void init(final ProcessorContext context) {
+                    state = (KeyValueStore<Integer, Integer>) context.getStateStore(stateStoreName);
+                }
+
+                @Override
+                public KeyValue<Integer, Integer> transform(final Integer key, final Integer value) {
+                    state.putIfAbsent(key, value);
+                    final KeyValue<Integer, Integer> result = new KeyValue<>(key, value);
+                    return result;
+                }
+
+                @Override
+                public void close() {}
+            }, stateStoreName)
+            .to(OUTPUT_TOPIC);
+
+        final Topology topology = builder.build();
+        final KafkaStreams client1 = new KafkaStreams(topology, streamsConfiguration());
+        final KafkaStreams client2 = new KafkaStreams(topology, streamsConfiguration());
+
+        final boolean[] client1IsOk = {false}; // has to be a final array, otherwise flag cannot be modified in lambda
+        client1.setStateListener((newState, oldState) -> {
+            if (newState == State.RUNNING &&
+                client1.localThreadsMetadata().iterator().next().standbyTasks().isEmpty()) {
+
+                client1IsOk[0] = true;
+            }
+        });
+        final boolean[] client2IsOk = {false}; // has to be a final array, otherwise flag cannot be modified in lambda
+        client2.setStateListener((newState, oldState) -> {
+            if (newState == State.RUNNING &&
+                client2.localThreadsMetadata().iterator().next().standbyTasks().isEmpty()) {
+
+                client2IsOk[0] = true;
+            }
+        });
+
+        client1.start();
+        client2.start();
+
+        TestUtils.waitForCondition(
+            () -> client1IsOk[0] && client2IsOk[0],
+            30 * 1000,
+            "At least one client is not in state RUNNING or has a stand-by task");
```

Comment:
I tried it and the test became flaky. The reason is that without fix the client is in `RUNNING` when the `IllegalStateException` is thrown, then it changes to `ERROR`, `PENDING_SHUTDOWN`, and finally `NOT_RUNNING`. It could happen that the wait condition gets satisfied when the client is in RUNNING. Then when the tests verifies `localThreadMetadata` two scenario can occur: 
1) client is still in RUNNING: `localThreadMetadata` contains a stand-by task -> assertion not satisfied 
2) client is not in RUNNING: exception is thrown because `localThreadMetadata` can only be accessed when the client is in `RUNNING`.
Both scenarios would let the test fail which would be OK, but not really clean. However, the test was sometimes green, which I currently do not understand. My guess would be race condition.
My approach checks 'localThreadMetadata' when the state changes to `RUNNING`, which should be safe in the error case. In the good case the test could run into the timeout if start-up is slow, though.